### PR TITLE
You can no longer insert beakers and syringes into cake

### LINKED ISSUE #2583: Using a syringe or open beaker on a cake will both use the item AND put it in the cake.
The beaker can be solved by putting a lid on it, but syringes are doomed to only ever have 10 units of whatever in them, with the other 5 in the cake.

The reason I include beakers in this is because I thought I'd be able to treat the cake like a bag of flour or other container and pour a ton of rum into it from the beaker. Because rum cake. As it stands, splashing a cake with the contents of a beaker does nothing but put the beaker into the cake, so you have to feed it a syringe for each unit of whatever you want in each slice.

Beaker-to-cake fluid transfer would be ideal.

### PR DESCRIPTION
**Pull request by Crazylemon64**
This being due to their after-attack being semi-co-opted by the cake's attackby - this prevents the insertion so usage of beakers and syringes doesn't do 2 actions at once, when you may have intended only 1.

Fixes #2583 (kind of)

:cl:Crazylemon
bugfix: Syringes now only inject/withdraw from cake, instead of being inserted in the cake in the process. Likewise, beakers will only splash, and not be inserted.
bugfix: Cake is no longer a bluespace container.
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Would you mind making it so you can only insert *one* item into a cake? You can literally shove unlimited amounts of small items into a cake, during it into a bluespace pocket that can be sliced open in a moments notice.

"lemme just get my ammo here" 

*slices cake->100 mags fall out*

**Comment by Crazylemon64** [comment]
I'm half-tempted to leave that in, JUST for that mental image of serious people slicing a cake open for loads of bullets, but alright, I'll add that limit.

**Comment by Fox-McCloud** [comment]
@Crazylemon64 I'm going to shank you!

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -131,9 +131,22 @@
...
 	else if(W.w_class <= 2 && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))
+		// ow this is bad
+		// In a nice world I'd have it check for things with an afterattack, but there's no way to detect that soooo
+		var/list/uninsertable_types = list(/obj/item/weapon/reagent_containers/syringe, /obj/item/weapon/reagent_containers/glass)
+		for(var/type in uninsertable_types)
+			if(istype(W, type))
+				return 1
+		if(contents.len > 1)
+			// No more bluespace cake
+			to_chat(user, "<span class='warning'>Something is already in the [src]!</span>"
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Whyyyyyy @Crazylemon64 whhyyyyyyy

Comment (Crazylemon64) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: aaaaaaaaaaaaaaaaagh rparens

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -131,9 +131,22 @@
 		)
 		inaccurate = 1
 	else if(W.w_class <= 2 && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))
+		// ow this is bad
+		// In a nice world I'd have it check for things with an afterattack, but there's no way to detect that soooo
+		var/list/uninsertable_types = list(/obj/item/weapon/reagent_containers/syringe, /obj/item/weapon/reagent_containers/glass)
+		for(var/type in uninsertable_types)
+			if(istype(W, type))
+				return 1
+		if(contents.len > 1)
```
Comment (tigercat2000) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: just do contents.len you madman

Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: That's what I told him =p

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -131,9 +131,22 @@
 		)
 		inaccurate = 1
 	else if(W.w_class <= 2 && istype(src,/obj/item/weapon/reagent_containers/food/snacks/sliceable))
+		// ow this is bad
+		// In a nice world I'd have it check for things with an afterattack, but there's no way to detect that soooo
+		var/list/uninsertable_types = list(/obj/item/weapon/reagent_containers/syringe, /obj/item/weapon/reagent_containers/glass)
+		for(var/type in uninsertable_types)
+			if(istype(W, type))
+				return 1
+		if(contents.len >= 1)
```
Comment (tigercat2000) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: :eyes:

**Comment by Dumbdumn5** [comment]
NUUUUUU! MY BLUESPACE SLICEABLES!HJGBFDGSVADX

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 3 above)
Comment (monster860) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: :eyes:

:eyes:

:eyes:

AAAAAAAAAAAA

**Comment by taukausanake** [comment]
Can we have a bluespace cake then for that sole purpose? Maybe chef antag item if they don't already? I'd rather just have a bluespace cake possible to bake

**Comment by fludd12** [comment]
I agree, a bluespace cake without the item restriction would be nice. Eating it should warp you a single tile in a random direction, too.

Baked using bluespace crystals instead of dough, same recipe as a regular cake?

**Comment by Crazylemon64** [comment]
Wehhhhh

**Pull request closed without merging**